UT4MasterServer/Models/StatisticBase.cs

```diff
@@ -516,109 +516,130 @@ public List<string> Validate()
+		if (BioRifleKills.HasValue && (!Kills.HasValue || BioRifleKills.Value > Kills.Value))
 		{
 			flaggedFields.Add(nameof(BioRifleKills));
+			flaggedFields.Add(nameof(Kills));
 		}
 		// Grenade Launcher kills are reported if they are more than the actual kills
-		if (BioLauncherKills.HasValue && Kills.HasValue && BioLauncherKills.Value > Kills.Value)
+		if (BioLauncherKills.HasValue && (!Kills.HasValue || BioLauncherKills.Value > Kills.Value))
 		{
 			flaggedFields.Add(nameof(BioLauncherKills));
+			flaggedFields.Add(nameof(Kills));
 		}
 		// Shock Beam kills are reported if they are more than the actual kills
-		if (ShockBeamKills.HasValue && Kills.HasValue && ShockBeamKills.Value > Kills.Value)
+		if (ShockBeamKills.HasValue && (!Kills.HasValue || ShockBeamKills.Value > Kills.Value))
 		{
 			flaggedFields.Add(nameof(ShockBeamKills));
+			flaggedFields.Add(nameof(Kills));
 		}
 		// Shock Core kills are reported if they are more than the actual kills
-		if (ShockCoreKills.HasValue && Kills.HasValue && ShockCoreKills.Value > Kills.Value)
+		if (ShockCoreKills.HasValue && (!Kills.HasValue || ShockCoreKills.Value > Kills.Value))
 		{
 			flaggedFields.Add(nameof(ShockCoreKills));
+			flaggedFields.Add(nameof(Kills));
 		}
 		// Shock Combo kills are reported if they are more than the actual kills
-		if (ShockComboKills.HasValue && Kills.HasValue && ShockComboKills.Value > Kills.Value)
+		if (ShockComboKills.HasValue && (!Kills.HasValue || ShockComboKills.Value > Kills.Value))
 		{
 			flaggedFields.Add(nameof(ShockComboKills));
+			flaggedFields.Add(nameof(Kills));
 		}
 		// Link kills are reported if they are more than the actual kills
-		if (LinkKills.HasValue && Kills.HasValue && LinkKills.Value > Kills.Value)
+		if (LinkKills.HasValue && (!Kills.HasValue || LinkKills.Value > Kills.Value))
 		{
 			flaggedFields.Add(nameof(LinkKills));
+			flaggedFields.Add(nameof(Kills));
 		}
 		// Link Beam kills are reported if they are more than the actual kills
-		if (LinkBeamKills.HasValue && Kills.HasValue && LinkBeamKills.Value > Kills.Value)
+		if (LinkBeamKills.HasValue && (!Kills.HasValue || LinkBeamKills.Value > Kills.Value))
 		{
 			flaggedFields.Add(nameof(LinkBeamKills));
+			flaggedFields.Add(nameof(Kills));
 		}
 		// Minigun kills are reported if they are more than the actual kills
-		if (MinigunKills.HasValue && Kills.HasValue && MinigunKills.Value > Kills.Value)
+		if (MinigunKills.HasValue && (!Kills.HasValue || MinigunKills.Value > Kills.Value))
 		{
 			flaggedFields.Add(nameof(MinigunKills));
+			flaggedFields.Add(nameof(Kills));
 		}
 		// Minigun Shard kills are reported if they are more than the actual kills
-		if (MinigunShardKills.HasValue && Kills.HasValue && MinigunShardKills.Value > Kills.Value)
+		if (MinigunShardKills.HasValue && (!Kills.HasValue || MinigunShardKills.Value > Kills.Value))
 		{
 			flaggedFields.Add(nameof(MinigunShardKills));
+			flaggedFields.Add(nameof(Kills));
```

Comment:
Maybe add a helper method, for example `AddKillsFor` to reduce code duplication

Reply:
This part is adding only name of the property to the list of flagged properties. I am not sure for what exactly would you add extension method?

Reply:
To reduce code. I updated my comment with code example

Reply:
You are not going to reduce anything. You will replace this:

![image](https://user-images.githubusercontent.com/12861786/213927668-317bb67c-799b-4ebb-ae65-89e28779b310.png)

with this:
![image](https://user-images.githubusercontent.com/12861786/213927681-1cfdd3d1-3596-406d-a1f7-37fabe4bb8ad.png)

Reply:
If the line count is important, I could do it like this:
![image](https://user-images.githubusercontent.com/12861786/213927973-56422654-cdcf-4b2f-901d-6f976c40bd72.png)

Reply:
But IMHO using `AddRange` will be better for code optimization.

![image](https://user-images.githubusercontent.com/905878/213929336-179832a5-8fa2-4a7a-9007-b83f0834338f.png)

(Remember to add comma at the end for better git diff view and PR code review experience in the future

Reply:
The problem with AddRange is that you have to instantiate unnecessary array/list each time. Using Add in multiple lines doesn't allocate anything additional.

Reply:
How about just having something like this (provided just as pseudocode):
```cs
	public void ValidateProperty(string name, int? value)
	{
		if (value.HasValue && Kills.HasValue && value.Value > Kills.Value)
		{
			flaggedFields.Add(name);
			flaggedFields.Add(nameof(Kills));
		}
	}
```

Reply:
This would reduce number of lines significantly imo

Reply:
Then create an array with all name/value pairs and just loop though all to reduce perhaps even more

Reply:
Not all rules are the same. I wouldn't make multiple methods out of one just for the sake of lines.

Reply:
All weapon stats seem the same. Having ~5 extra methods with different conditions is definitely something i'd prefer over this long unreadable mess. Its not for the sake of lines, its for the sake of maintainability and ease of reading.

Reply:
If that is what you prefer, I will extract similar logic to separate methods.

Reply:
I run a simple BenchmarkDotNet benchmark and `DoubleAdd` is better

```
|    Method |       Mean |     Error |     StdDev |     Median |        Min |         Max | Ratio | RatioSD | Allocated | Alloc Ratio |
|---------- |-----------:|----------:|-----------:|-----------:|-----------:|------------:|------:|--------:|----------:|------------:|
| DoubleAdd |   510.7 ns |  68.49 ns |   929.4 ns |   400.0 ns |   300.0 ns | 24,700.0 ns |  1.00 |    0.00 |     640 B |        1.00 |
|  AddRange | 2,627.3 ns | 187.31 ns | 2,541.9 ns | 2,100.0 ns | 1,500.0 ns | 25,600.0 ns |  6.21 |    5.98 |     680 B |        1.06 |
```

Benchmark code:

```C#
using BenchmarkDotNet.Attributes;
using BenchmarkDotNet.Engines;
using BenchmarkDotNet.Running;

namespace Benchmark;

[SimpleJob(RunStrategy.Monitoring, launchCount: 2, warmupCount: 5, iterationCount: 1000)]
[MemoryDiagnoser]
[MinColumn, MaxColumn, MedianColumn]
public class AddTwiceVsAddRange
{
	public int? Wins { get; set; }
	public int? Losses { get; set; }

	private List<string> flaggedFields;

	[GlobalSetup]
	public void GlobalSetup()
	{
		flaggedFields = new List<string>();
	}

	[Benchmark(Baseline = true)]
	public List<string> DoubleAdd()
	{
		flaggedFields.Add(nameof(Wins));
		flaggedFields.Add(nameof(Losses));
		return flaggedFields;
	}

	[Benchmark]
	public List<string> AddRange()
	{
		flaggedFields.AddRange(new[]
		{
			nameof(Wins),
			nameof(Losses),
		});

		return flaggedFields;
	}
}

public class Program
{
	public static void Main(string[] args)
	{
		var summary = BenchmarkRunner.Run<AddTwiceVsAddRange>();
	}
}
```

Reply:
Extracted logic to multiple methods.